contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,212 @@
+processing logic. This was fixed in December 2020, and this fix reportedly
+reduced the systemd mount-scanning-related overhead by 30-50%. The second is a
+lack of granularity in the events the kernel sends to systemd about mountpoint
+changes, which necessitates a full rescan of all mountpoints even on a minor
+change or single mountpoint addition. The kernel change to remedy this seems
+unlikely to move forward in the near term.  (See
+[BZ1819869](https://bugzilla.redhat.com/show_bug.cgi?id=1819868) for more
+details)
+
+For example, a fairly small OpenShift deployment today (which does not yet have
+the systemd December 2020 fix) has upwards of 400 mountpoints on every node
+owned by Kubelet, the Container Runtime, and infrastructure containers like OVN
+or CSI operators. Hiding the container mountpoints reduces the steady-state
+systemd CPU utilization from ~10% to ~0% on every worker node. On a more
+heavily loaded system, with 1000’s of container mountpoints, systemd’s usage
+will cap out at 100% of a single CPU core. Even with the 30-50% reduction in
+systemd usage expected from the December 2020 fix, having systemd take up as
+much as 50-70% of a single CPU in the worst-case scenario is a lot of
+unnecessary overhead.
+
+Administrators of other OSes with other host OS services that monitor or
+process mountpoints indiscriminately could also benefit from hiding these
+container-specific mountpoints from the OS at large.
+
+### Tidiness and Encapsulation
+
+These container overlay mounts, secret and configmap mounts, and namespace
+mounts are all internal implementation details around the way the Kubernetes
+and the Container Runtime allocate resources to containers. For normal
+day-to-day operation, there is no need for these special-use mounts to be
+visible in the host OS's top-level 'mount' or 'findmnt' output.
+
+### Security
+
+While access to container-specific mounts in Kubernetes (like secrets) today
+are protected by filesystem permissions, the fact that the locations are
+visible to unprivileged users via the 'mount' command can provide information
+which could be used in conjunction with other vulnerabilities to decipher
+secrets or other protected container mount contents. Locking access to even
+listing the mountpoints behind the ability to enter the mount namespace (which
+in turn requires `CAP_SYS_CHROOT` and `CAP_SYS_ADMIN`) removes one level of
+potential access to container-specific mounts.
+
+### No one uses it (that we know of)
+
+After much investigation and conversation on both the sig-node and sig-storage
+mailing lists, as well as within Red Hat's OpenShift communities, we have yet
+to turn up a use case that relies on this container-to-host propagation.
+
+### No one _should_ use it
+
+In fact, it may be seen as an anti-pattern and run contrary to the
+container-native philosophy. Relying on a host OS to interact with a
+container-owned mountpoint seems suspect.
```

Comment:
It's not clear to me from this paragraph why this is an anti-pattern. I'd suggest we either drop this paragraph or substantiate this a bit better :)

Reply:
Sure, never hurts to try to clarify my assumptions here...

So I was basically thinking that the cloud-native way of doing things is to make sure an end-user's full solution works by being distributed in a set of containers.  It seems obvious to me that requiring a special package on the Host OS to interact with something your containers do is not cloud-native.  Do you agree?  Can you think of a better way I could say that?

Reply:
I don't think this is obvious, and I'd shy away from making universal recommendations of good or bad, because there's always an exception to break the rule... I'd just remove this section, I think :)